### ISSUE TITLE
Speedgoose, TypeError: Cannot read properties of null (reading '_id')

### ISSUE THREAD

**Issue by IamSinghMohit** [arqo123/speedgoose#84]
i don't know why but it speedgoose is giving me typeerror that cannot read properties of id when trying query a single document in a collection. It is working fine if collection is not empty and it does not throw any errors but in case of empty collection it is throwing me error.

```ts
    static async stats(
        req: Request<TGetProductSchema>,
        res: Response,
        next: NextFunction,
    ) {
        try {
            const product = await ProductModel.findOne({})
                .sort({ price: -1 })
                .limit(1)
                .cacheQuery();
            ResponseService.sendResponse(res, 200, true, {
                max_price: (product?.price || 0) + 10,
            });
        } catch (error) {
            console.log(error);
        }
    }
```

this is the whole error 
```
TypeError: Cannot read properties of null (reading '_id')
    at generateCacheKeyForSingleDocument (/home/madara/Documents/main/server/node_modules/speedgoose/lib/src/utils/cacheKeyUtils.js:22:30)
    at getHydratedDocument (/home/madara/Documents/main/server/node_modules/speedgoose/lib/src/utils/hydrationUtils.js:29:76)
    at hydrateResults (/home/madara/Documents/main/server/node_modules/speedgoose/lib/src/utils/hydrationUtils.js:57:123)
    at prepareQueryResults (/home/madara/Documents/main/server/node_modules/speedgoose/lib/src/extendQuery.js:19:101)
    at execQueryWithCache (/home/madara/Documents/main/server/node_modules/speedgoose/lib/src/extendQuery.js:34:16)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by arqo123** [comment]
To check, thanks for another catch!

**Comment by arqo123** [comment]
Fixed in speedgoose@2.0.12, please retest and close the ticket. 
Btw i've noticed that you're fetching the data streight from the db and passing it directly to the response. To speed up your app i sugesst You to use the `lean()` option to avoid unnecessary hydration (which might take a few ms per each document.

**Comment by IamSinghMohit** [comment]
Thanks for advice.
